api/pkg/gateway/context_test.go

```diff
@@ -0,0 +1,477 @@
+package gateway
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/labstack/echo/v4"
+
+	"github.com/shellhub-io/shellhub/pkg/api/authorizer"
+	"github.com/shellhub-io/shellhub/pkg/models"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRole(t *testing.T) {
+	cases := []struct {
+		description string
+		headers     map[string]string
+		expected    authorizer.Role
+	}{
+		{
+			description: "verify if given value returns from header",
+			headers: map[string]string{
+				"X-Role": authorizer.RoleOwner.String(),
+			},
+			expected: authorizer.Role("owner"),
+		}, {
+			description: "validate empty ID string, for ID behaviour",
+			headers: map[string]string{
+				"X-Role": "",
+			},
+			expected: authorizer.RoleInvalid,
+		}, {
+			description: "validate empty headers, for fail function",
+			headers:     map[string]string{},
+			expected:    authorizer.RoleInvalid,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.description, func(t *testing.T) {
+			e := echo.New()
+			req := httptest.NewRequest(http.MethodPost, "/", nil)
+
+			for k, v := range tc.headers {
+				req.Header.Set(k, v)
+			}
+
+			rec := httptest.NewRecorder()
+			ctx := e.NewContext(req, rec)
+
+			ctxNew := Context{
+				nil,
+				ctx,
+			}
+
+			role := ctxNew.Role()
+
+			require.Equal(t, tc.expected, role)
+		})
+	}
+}
+
+func TestTenant(t *testing.T) {
+	cases := []struct {
+		description string
+		headers     map[string]string
+		expected    *models.Tenant
+	}{
+		{
+			description: "verify if given value returns from header",
+			headers: map[string]string{
+				"X-Tenant-ID": "00000000-0000-4000-0000-000000000000",
+			},
+			expected: &models.Tenant{
+				ID: "00000000-0000-4000-0000-000000000000",
+			},
+		}, {
+			description: "validate empty ID string, for ID behaviour",
+			headers: map[string]string{
+				"X-Tenant-ID": "",
+			},
+			expected: nil,
+		}, {
+			description: "validate empty headers, for fail function",
+			headers:     map[string]string{},
+			expected:    nil,
+		},
```

Comment:
The same applies here. You can safely remove the first case.

Also, you can do the same in the other tests.